src/link/script.y

```diff
@@ -369,10 +390,47 @@ static void setAddr(uint32_t addr) {
 	} else {
 		pc = addr;
 	}
+	isPcFloating = false;
+}
+
+static void makeAddrFloating(void) {
+	auto const &context = lexerStack.back();
+	if (activeType == SECTTYPE_INVALID) {
+		scriptError(context, "Cannot make PC floating because no memory region is active");
+		return;
+	}
+
+	isPcFloating = true;
+	floatingAlignMask = 1;
+	floatingAlignOffset = 0;
 }
 
 static void alignTo(uint32_t alignment, uint32_t alignOfs) {
 	auto const &context = lexerStack.back();
+	if (activeType == SECTTYPE_INVALID) {
+		scriptError(context, "Cannot modify PC because no memory region is active");
+		return;
+	}
+
+	if (isPcFloating) {
+		if (alignment >= 16) {
+			setAddr(floatingAlignOffset);
+		} else {
+			uint32_t alignSize = 1u << alignment;
+
+			if (alignOfs >= alignSize) {
+				scriptError(context, "Cannot align: The alignment offset (%" PRIu32
+						      ") must be less than alignment size (%" PRIu32 ")\n",
+					    alignOfs, alignSize);
+				return;
+			}
+
+			floatingAlignMask = alignSize;
```

Comment:
I think what you have here is a "`floatingAlignSize`" -- a value like `1 << alignment` -- not an align *mask* (that would subtract 1, so you can `&` by it, not `%` by it).